### ISSUE TITLE
AppStore and `mas` aren’t in synch, AppStore still shows App as upgradable

### ISSUE THREAD

**Issue by antoine2711** [mas-cli/mas#642]
## Your Environment

- `mas version`: 1.8.7
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 14.7.1

### mas Install Method

- [X] `brew install mas` (homebrew-core)

## Describe the Bug

After upgrading an App with `mas upgrade`, the AppStore still list the App as an upgradable App.

## To Reproduce

1. `mas upgrade`
2. Nothing happens (see screenshot).

## Expected Behavior

AppStore shows the app as updated

## Actual Behavior

AppStore still propose to update the App.

## Screenshots, Terminal Output
![image](https://github.com/user-attachments/assets/5b56969c-237e-447d-acb8-924a9541748a)

**Comment by rgoldberg** [comment]
This seems to be the same issue as #387. It's also [noted in the README](https://github.com/mas-cli/mas?tab=readme-ov-file#-eventual-consistency).

What are the versions that the Mac App Store reports as being newer than what's already installed?

What does each app report its version as when it is running? Get from the menu for the app).

After that, the best thing to do is to:

- ensure that auto updates are disabled
- don't upgrade any of the affected apps via the Mac App Store or via mas for around a day (even better, don't use either of them for anything for the whole day)
- after the delay of around a day, check to see if the Mac App Store still thinks the apps can be updated but mas doesn't think they can be
- if that's still the case,reload the Updates page in the Mac App Store either via keyboard (Command-r) or mouse (menu > Store > Reload Page)

**Comment by rgoldberg** [comment]
@antoine2711 Did everything work fine after a delay?

**Comment by antoine2711** [comment]
Now sure. I think it went back to a previous version… But now, it’s working (updating worked, and after going to App Store not previouly opened, it didn’t ask to upgrade).

I did get this weird thing on another computer…

![image](https://github.com/user-attachments/assets/b705d1a5-e04f-4287-87ad-fcca9c9debd4)

The same App with TWO different versions…

Regards, Antoine

**Comment by antoine2711** [comment]
This is after upgrade…

![image](https://github.com/user-attachments/assets/f23ca683-f19c-4dab-9d31-12f94d17d80f)

**Comment by rgoldberg** [comment]
You almost certainly have multiple instances of the same app installed somewhere under `/Applications`, which is tracked in #462.

There are some diagnostic steps there to verify that this is the problem.

We are currently limited by the Apple code we call to only updating one of the duplicate apps. Simplest current solution is to manually delete the extra app installations from the file system.

**Comment by rgoldberg** [comment]
@antoine2711 It looks like all of the problems brought up in this issue are already tracked elsewhere.

If that's not the case, please let me know what we remains untracked elsewhere.

If all tracked elsewhere, please let me know so I can close out this issue as a duplicate.

Thanks.

**Comment by antoine2711** [comment]
@rgoldberg :
It seem to be a bit different.

I have never installed twice the same app. In any case, you can see that the name is the same.
I have a question: how can I get the path to an installed app?

Now, without going into AppStore, I can’t upgrade an app…

```
mas outdated
1295203466 Windows App (11.0.3 -> 11.0.7)
 03:58:33 antoine@MacMini5-AB-2019-i7.local:/usr/local/dev/antoine2711/client-cult-xojo-test/src/Noyau 
mas list
640199958   Developer             (10.7)
409183694   Keynote               (14.2)
409907375   Remote Desktop        (3.9.7)
1474276998  HP Smart              (17.30.1)
1435575700  DirEqual              (6.4)
1369524274  QR Capture            (2.6.10)
1295203466  Windows App           (11.0.7)
497799835   Xcode                 (16.1)
409201541   Pages                 (14.2)
450664466   NAS Navigator2        (3.11)
408981434   iMovie                (10.4.3)
472995993   Remote Desktop - VNC  (1.1)
1295203466  Windows App           (11.0.3)
409203825   Numbers               (14.2)
1024640650  CotEditor             (5.0.5)
1037126344  Apple Configurator    (2.17)
967004861   HP Easy Scan          (2.0.0)
 03:58:41 antoine@MacMini5-AB-2019-i7.local:/usr/local/dev/antoine2711/client-cult-xojo-test/src/Noyau 
mas upgrade 1295203466
Upgrading 1 outdated application:
Windows App (11.0.3) -> (11.0.7)
==> Downloaded Windows App (11.0.7)
==> Installing Windows App (11.0.7)
==> Installed Windows App (11.0.7)
 03:59:12 antoine@MacMini5-AB-2019-i7.local:/usr/local/dev/antoine2711/client-cult-xojo-test/src/Noyau 
mas outdated
1295203466 Windows App (11.0.3 -> 11.0.7)
```

Regards, Antoine

**Comment by rgoldberg** [comment]
This probably is occurring because Microsoft changed the name of the app from "Microsoft Remote Desktop" to "Windows App".

The solution might be to, in `App Store.app`:

- open the Account page (accessible by clicking on the text at the bottom of the left pane that says the name of the Apple ID for which you're signed in in the Mac App Store)
- check to see if either `Microsoft Remote Desktop` and/or `Windows App` is listed amongst your Purchased apps (possibly one or more times each, so look through the whole list without short-circuiting the first time you find them)
- try to reload the page via the keyboard (Command-r) or via the menu (Store > Reload Page)
- try `mas outdated` again

If that doesn't fix everything, try a dry run of an uninstall to see the paths to all the apps with the same app ID as `Windows App`:

```shell
sudo mas uninstall --dry-run 1295203466
```

This will NOT uninstall anything. `sudo` is required, though, even for a dry run.

**Comment by antoine2711** [comment]
@rgoldberg :
it gave me this:
![image](https://github.com/user-attachments/assets/c060253d-1167-4b81-802d-f3efba9de315)

So, the problem is with Apple’s API, and not our `mas` command tool, I guess?
Thanks for the info!

Regards, Antoine

**Comment by rgoldberg** [comment]
I assume the latter path doesn't exist on the file system, right?

If so, the problem is from the Apple API.

Did you try to "Reload Page"?

**Comment by rgoldberg** [comment]
So the issue is duplicate apps. Will close.

Feature request #462 already exists for the ability to update multiple apps with the same ID.

**Comment by antoine2711** [comment]
@rgoldberg :

Should I write a new issue to have the `mas info ID` give the path (or paths if more than one) of the applications installed?

Can you remove the label “awaiting user reply”?

Regards, Antoine

**Comment by rgoldberg** [comment]
@antoine2711 Thanks for the offer of making an issue, but it shouldn't be necessary because:

#660 is an umbrella issue for detecting & potentially fixing odd/bad states for MAS apps. Duplicate app IDs are the first (& so far only) state mentioned in #660. Outputting app paths is included in the issue as an example of one way of highlighting potential underlying issues like duplicate app IDs.

#626 is an umbrella issue for refactoring the whole output system (it is somewhat terse, as I wanted to document quickly, and only flesh it out it others really need detailed explanations). I would like to change specific output only after the whole system has been refactored.

I'm the only currently active mas developer, so I don't think we'll need a separate issue for app paths.

Before I refactor the output system, I need to:

- finish refactoring our release process
- fix other administrative issues
- finish fixing bugs in Swift Argument Parser shell completion script generation
- implement mas shell completions for zsh, bash & fish
- possibly complete other tasks

I also want to discuss various large initiatives (like release & output refactoring) & establish task priorities with the project lead (who doesn't have much time to work on mas due to other commitments), so it might take time to get to this all.

**Comment by antoine2711** [comment]
@rgoldberg : thanks for taking the time to explain all of this to me.

Keep up the good work, and I’ll continue to do my testing.

Regards, Antoine

**Comment by rgoldberg** [comment]
@antoine2711 Thanks for using mas, and for taking the time to report & investigate issues.

If you experience other problems, or have feature requests, please let us know.

**Comment by antoine2711** [comment]
Just a follow-up.

I deleted `file '/Applications/Microsoft Remote Desktop.app'`

And then, I tried:
```
mas outdated
1295203466 Windows App (11.0.3 -> 11.0.7)
```
I did:
```
mas upgrade 
Upgrading 1 outdated application:
Windows App (11.0.3) -> (11.0.7)
==> Downloaded Windows App (11.0.7)
==> Installing Windows App (11.0.7)
==> Installed Windows App (11.0.7)
```
But it’s still in the outdated, and can be forever upgraded. I’ll retry in 24h

Regards, Antoine

**Comment by rgoldberg** [comment]
@antoine2711 Can you rerun the dry run of the uninstall (because the Apple data source might still think that the app you deleted is still installed)?

```shell
sudo mas uninstall --dry-run 1295203466
```

If the deleted path is still listed by that command, you should reload page in the Mac App Store GUI as [mentioned above](https://github.com/mas-cli/mas/issues/642#issuecomment-2505886305).

If not listed, try upgrading after waiting the 24 hours you mentioned. If that doesn't fix it, we'll have to investigate more.

**Comment by antoine2711** [comment]
> @antoine2711 Can you rerun the dry run of the uninstall (because the Apple data source might still think that the app you deleted is still installed)?
> If the deleted path is still listed by that command, you should reload page in the Mac App Store GUI as [mentioned above](https://github.com/mas-cli/mas/issues/642#issuecomment-2505886305).

Well, I deleted the application in the Finder.

> If not listed, try upgrading after waiting the 24 hours you mentioned. If that doesn't fix it, we'll have to investigate more.

I waited 24h, and now, outdated shows me nothing, so I guess it was Apple data source that didn’t realized I deleted the app in the Finder.

Regards, Antoine
